### PR TITLE
Add check for LP#2029857

### PR DESCRIPTION
**Pull request by wilkmar**
This PR adds detection support for [LP#2029857](https://bugs.launchpad.net/charm-octavia/+bug/2029857)

### CODE REVIEW

File: hotsos/defs/scenarios/openstack/octavia/bugs/lp2029857.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+''ovn'' was not found\.)'
+  amphora_provider_enabled:
+    config:
+      handler: 'hotsos.core.plugins.openstack.OpenstackConfig'
+      path: 'etc/octavia/octavia.conf'
+      assertions:
+        - section: api_settings
+          key: enabled_provider_drivers
+          ops: [[contains, amphora]] 
+  ovn_provider_enabled:
```
Comment (pponnuvel) [inline hotsos/defs/scenarios/openstack/octavia/bugs/lp2029857.yaml:17]: This is fine as-is but you can combine these two checks as:
```
  amphora_ovn_provider_enabled:
    config:
      handler: 'hotsos.core.plugins.openstack.OpenstackConfig'
      path: 'etc/octavia/octavia.conf'
      assertions:
        - section: api_settings
          key: enabled_provider_drivers
          ops: [[contains, amphora]]
        - section: api_settings
          key: enabled_provider_drivers
          ops: [[contains, ovn]]
```

Ideally it'd look like (this one doesn't work):
```
  amphora_ovn_provider_enabled:
    config:
      handler: 'hotsos.core.plugins.openstack.OpenstackConfig'
      path: 'etc/octavia/octavia.conf'
      assertions:
        - section: api_settings
          key: enabled_provider_drivers
          varops: [[contains, amphora], [contains, ovn]]
```

Maybe @dosaboy  knows how to combine multiple 'contains' into one.

**Review by pponnuvel — APPROVED** [review]
Looks good @wilkmar! One minor comment.

File: hotsos/defs/scenarios/openstack/octavia/bugs/lp2029857.yaml (hunk 1 above)
Comment (wilkmar) [reply hotsos/defs/scenarios/openstack/octavia/bugs/lp2029857.yaml]: done (opt 1)

File: hotsos/defs/scenarios/openstack/octavia/bugs/lp2029857.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      - has_2029857
+      - amphora_provider_enabled
+      - ovn_provider_enabled 
+    raises:
+      type: LaunchpadBug
+      bug-id: 2029857
+      message: >-
+        Octavia is configured to support both amphora and ovn provider
+        drivers but it is impacted by a known bug that is causing octavia
+        api errors. See LP#2029857 for more information.
```
[the remark this answers is not in the document]
Comment (wilkmar) [reply hotsos/defs/scenarios/openstack/octavia/bugs/lp2029857.yaml]: done

**Review by pponnuvel — APPROVED** [review]

**Review by brianphaley — APPROVED** [review]

**Pull request merged**